src/libraries/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -109,14 +109,45 @@ public string Value
         ///   A return value indicates whether the conversion succeeded.
         /// </summary>
         /// <param name="value">
-        ///   When this method returns, contains the see cref="DateTime"/> value equivalent of the text contained in this instance,
+        ///   When this method returns, contains the <see cref="DateTime"/> value equivalent of the text contained in this instance,
         ///   if the conversion succeeded, or zero if the conversion failed.
         /// </param>
         /// <returns>
         ///  <see langword="true"/> if instance was converted successfully;
         ///  otherwise, <see langword="false"/>
         /// </returns>
-        public bool TryGetDateTime(out DateTime value) => JsonHelpers.TryParseAsISO(_value.AsSpan(), out value);
+        public bool TryGetDateTime(out DateTime value)
+        {
+            ReadOnlySpan<char> span = _value.AsSpan();
+
+            if (!JsonHelpers.IsValidDateTimeOffsetParseLength(span.Length))
+            {
+                value = default;
+                return false;
+            }
+
+            bool isQuoted = span[0] == JsonConstants.Quote && span[span.Length - 1] == JsonConstants.Quote;
+
+            int length = JsonReaderHelper.GetUtf8ByteCount(span);
+
+            Span<byte> bytes = length <= JsonConstants.StackallocThreshold
+                ? stackalloc byte[JsonConstants.StackallocThreshold]
+                : new byte[length];
+
+            JsonReaderHelper.GetUtf8FromText(span, bytes);
+
+            // trim beginning and ending quotes
+            bytes = isQuoted ? bytes.Slice(1, length - 2) : bytes.Slice(0, length);
+
+            if (bytes.IndexOf(JsonConstants.BackSlash) != -1)
+            {
+                return JsonReaderHelper.TryGetEscapedDateTime(bytes, out value);
+            }
+
+            Debug.Assert(bytes.IndexOf(JsonConstants.BackSlash) == -1);
+
+            return JsonHelpers.TryParseAsISO(bytes, out value);
```

Comment:
I would follow the pattern here to avoid returning non-default `DateTime` value as the out when we return false.

https://github.com/dotnet/runtime/blob/24062e30d77e5b0fbc1c3b518bd76a8edee09478/src/libraries/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs#L856-L863